### PR TITLE
Fixes #9104

### LINKED ISSUE #9104: Diona Nymphs freeze on ventcrawl
@DrCelt
@DrCelt
@DrCelt

Even when crawling into a 20C pipe filled with air, they freeze nearly instantly
This also affects monkeys and other vent crawling entities, such as xenos.

"According to Celt this is a feature because more air pressure = more heat transfer so you freeze to 20c ;)"

Last issue was locked and I cannot imagine why.

### LINKED ISSUE #10196: Revert Celt's buggy shitcode hypothermia
Too many bugs, too many issues, none of it's getting fixed, Celt took the money he got for coding it and ran (all $0.02)

For example, if you try to ventcrawl in a room temperature vent, you will freeze to death.

Raise the temperature to above room temperature and average pressure (101.3kpa) in the vents? Freeze to death still

Unwrench a pipe with high pressure? You bet your ass it's gonna freeze you to death

Hypothermia has an issue of working properly, and I've even gotten hypothermia sleeped with no warning messages, while in above average temperature in vents. This issue has been reported by dionas for ages but never touched. Remove shitcode!

### PR DESCRIPTION
**Pull request by ghost**
fixes #9104
closes #10196
https://www.youtube.com/watch?v=K9q70zCJYBc

### CODE REVIEW

**Comment by ghost** [comment]
Is this the beginning of the 25 bug fix challenge?

**Comment by ghost** [comment]
No, this doesn't count.

**Comment by ghost** [comment]
I'll do that later

**Comment by ComicIronic** [comment]
I don't support this way of doing it: you haven't avoided the issue of high pressure amplifying heat transfer. I'd rather you rework the pressure impact to change that instead.

**Comment by ghost** [comment]
@ComicIronic That's a non-issue. High pressure DOES amplify heat transfer.

File: code/modules/mob/thermoregulation.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@
 	if(loc_temp < bodytemperature)
 		// We're going to try and just use exposed area(temperature difference)/cold divisor, and assume we're only conducting.
 		var/thermal_loss = (1-get_cold_protection())  				// How much of your skin is exposed.
-		if(environment.total_moles > MOLES_CELLSTANDARD || !IS_SPACE_COLD)
+		if(!isVentCrawling() && (environment.total_moles > MOLES_CELLSTANDARD || !IS_SPACE_COLD))
```
Comment (9600bauds) [inline code/modules/mob/thermoregulation.dm]: He did it again.
The absolute madman, he did the exact same thing again.
https://github.com/d3athrow/vgstation13/pull/9116

Comment (ghost) [inline code/modules/mob/thermoregulation.dm]: Nope! this time it's the correct proc

**Comment by ghost** [comment]
> Natural convection heat transfer from a horizontal
> cylinder is affected by air pressure. Higher air pressure
> causes better heat transfer and reduces heat transfer
> resistance. The steady temperatures of test specimen show
> this statement where for example at same input power the
> steady temperature decreases from 196 to 120 by changing of
> pressure from 18 to 220,000 Pa. As these tests have been
> done in different input powers but in each pressure results
> had dependency to temperature and different input powers
> had same results.
> In tests pressure range the convection coefficient can
> become ten times more only by changing air pressure. Also
> the pressure changes cause changes in ratio of heat transfer
> role between radiation and convection. In rarefied air,
> radiation has the priority but at high pressure this priority
> lessened and convection becomes more important. For
> example at 60 Celsius temperature, although the radiation
> coefficient is same on different pressures but the convection
> coefficient becomes 7 times more on 220 kPa than 18 Pa. this
> rhythm repeats on different temperatures.

**Comment by 9600bauds** [comment]
It doesn't amplify your heat transfer though. It amplifies chillness.
It specifically ONLY LOWERS your temperature. High amount of extremely hot moles doesn't heat you up faster.
Low pressure doesn't reduce your heat transfer either.

This is a VERY SPECIFIC KLUDGE THAT ONLY AFFECTS HYPOTHERMIA

**Comment by ghost** [comment]
> Low pressure doesn't reduce your heat transfer either.

Blame cold-space.

**Comment by 9600bauds** [comment]
I also really don't see the point in this feature and it has never ever mattered in all these months, besides walking inside a gas miner without a hardsuit or very very very very very very specific distro sabotage to make people without a hardsuit instantly fall asleep (which is analogous to N2Ofludd or just coldfludd if this were removed) which has never happened anyways

**Comment by ghost** [comment]
@9600bauds What an awful line of reasoning. What other true-to-life things should we remove because they don't matter?

**Comment by 9600bauds** [comment]
I'm saying that, WHEN IT MATTERS, it's awful, even if it's almost never. Don't even need to delve into this whole "true to life!" thing.

**Comment by ComicIronic** [comment]
Is thermal loss actually capped by the temp difference, or can you lose more temp than you actually need to equalise?

**Comment by ghost** [comment]
@ComicIronic your body temperature can never go lower than the ambient temperature via external heat loss.

**Comment by ComicIronic** [comment]
Then what exactly is this fixing?

**Comment by ghost** [comment]
It's technically a tweak rather than a bug fix, actually, but the issue is considered to be a bug so I named it as such - in essence, in 20°c temperature at a high pressure body temperature would drop to 20°c, which is undesirable due to the more realistic temperature system which has body temperature best at 37°c, as is true in muh real life.

**Comment by ComicIronic** [comment]
Right, but the environment doesn't normally give you hypothermia just because the ambient is at 20, hypothermia happens when your CORE drops. Just being exposed to 20 Celsius isn't going to kill you.

**Comment by ghost** [comment]
Correct. But if you're exposed to 20°c at a super super high pressure you'll lose core temperature fairly quickly. That's what the issue in the OP is about.

**Comment by clusterfack** [comment]
Just remove the pressure multiplier (or cap it at 1 atm)

**Comment by ghost** [comment]
I don't think capping it at one is the right idea, but I could decrease the multiplier?

**Comment by 9600bauds** [comment]
Capping it at 1 would mean just removing the check because the multiplier can't even go below 1

**Comment by ghost** [comment]
That's true

**Comment by clusterfack** [comment]
Good then remove the multiplier, it serves no purpose

**Comment by ghost** [comment]
Feck off Cluster, its purpose is to at least attempt to accurately simulate heat transfer.

**Comment by clusterfack** [comment]
Is it actually doing that though? For one thing that is VERY questionable.

And for another its fucking unfun and has added nothing in its current implementation.

**Comment by ghost** [comment]
@clusterfack What are you talking about re pt 1? 

re pt 2 what else should we remove that's real but unfun?

**Comment by 9600bauds** [comment]
Anything that's unfun should go at earliest notice
It being "real" is not redeeming by itself

**Comment by ghost** [comment]
@9600bauds aight, not being able to host a slumber party in space is pretty unfun. Remove space doing damage when

**Comment by 9600bauds** [comment]
But you can do that.
The Holodeck has a "space" setting :^)

**Comment by ghost** [comment]
:^)

**Comment by ComicIronic** [comment]
> trying to be realistic
> makes a pr where pipe magic stops heat transfer

**Comment by ghost** [comment]
@ComicIronic *shrug everyone called it a bug and told me to fix it.

**Comment by ComicIronic** [comment]
I don't call it a bug, I call it a design flaw. This fix is also a design flaw. The whole pressure system is flawed for the way it interacts with everything else.

**Comment by ghost** [comment]
The _pressure_ system is flawed, not the way cold deals with it.

Unless you want a couple hundred lines of pressure buffs to fix this, this is the way forward.

**Comment by clusterfack** [comment]
It does not look like its scaling linearly with pressure at all there celt

**Comment by ghost** [comment]
@clusterfack fair enough - so, as I said, "I don't think capping it at one is the right idea, but I could decrease the multiplier?"

**Comment by clusterfack** [comment]
I think that would be for the best

**Comment by ghost** [comment]
aight, will do.

**Comment by ghost** [comment]
seems to matter most as a difference for low pressures. and then decrease progressively

THANKS COLD SPACE.

**Comment by clusterfack** [comment]
And tbh thats only for temperature loss, is it really the same for temperature increases?

**Comment by ghost** [comment]
temperature loss is the same as temperature increase if you think about it

**Comment by clusterfack** [comment]
I disagree

**Comment by clusterfack** [comment]
Completely

**Comment by ghost** [comment]
After pressure_diff = 1.5 (x in the diagram), this will now be the non linear progression: http://www.wolframalpha.com/input/?i=-4.5%2Fx+%2B+4.5

**Comment by clusterfack** [comment]
I bet celt fucked this up
